### DISCUSSION TITLE
v3.50~20

### DISCUSSION THREAD

**Discussion by BobvanderWeg** [freakent/dbus-mqtt-devices#130] [Q&A]
I'm using with succes the dbus-mqtt-devices for several purposes (EVcharger, Temperature, PVinverter)
Currently using the beta version v3.50 20 of Victron OS.
When upgrading to a higher version of Victron OS v3.50 32 or 40 the driver dbus-mqtt-devices seems not to start anymore telling the yaml driver is missing. 
When I perform a rollback to v3.50 20 all is working fine again. 
Anybody familiar with this issue? (Using a MultiPlus-II GX)

**Comment by freakent** [comment]
Good to hear you have this working on the MultiPlus-II GX. What's the actual error you are receiving? Is it the dataclasses error or the YAML library that is missing. Please cut and paste the error message if you can.

I would check the /data/rc.local file. There should be a line that executes the bin/setup-dependencies.sh script which ensure thes YAML library is installed on each reboot.

**Comment by BobvanderWeg** [comment]
Noticed the service is running now on V3.50~39.
Using MQTT I send messages to a temp sensor and found error mentioned below.
[VenisOS35039.txt](https://github.com/user-attachments/files/17556494/VenisOS35039.txt)
[VenisOS35039.txt](https://github.com/user-attachments/files/17556511/VenisOS35039.txt)


root@nanopi:/data# more /data/rc.local
/data/drivers/dbus-mqtt-devices-0.8.0/bin/setup-dependencies.sh
ln -s /data/drivers/dbus-mqtt-devices-0.8.0/bin/service /service/dbus-mqtt-devices
root@nanopi:/data#


root@nanopi:/data# tail -f /var/log/dbus-mqtt-devices/current | tai64nlocal
2024-10-29 12:02:51.763855500 INFO:device:**** Registering device: fe002, services: {'t1': 'temperature'} ****
2024-10-29 12:02:51.764349500 INFO:device:Registering Service temperature for client fe002
2024-10-29 12:02:51.764852500 INFO:device_service_config:About to open config file
2024-10-29 12:02:52.492365500 INFO:device_service:Registering service temperature for client fe002 at path com.victronenergy.temperature.mqtt_fe002_t1
2024-10-29 12:02:52.525084500 INFO:settingsdevice:Setting /Settings/Devices/mqtt_fe002_t1/ClassAndVrmInstance does not exist yet or must be adjusted
2024-10-29 12:02:52.601638500 WARNING:vedbus:USING OUTDATED REGISTRATION METHOD!
2024-10-29 12:02:52.602164500 WARNING:vedbus:Please set register=False, then call the register method after adding all mandatory paths. See https://github.com/victronenergy/venus/wiki/dbus-api
2024-10-29 12:02:52.607993500 INFO:vedbus:registered ourselves on D-Bus as com.victronenergy.temperature.mqtt_fe002_t1
2024-10-29 12:02:52.618698500 INFO:device_service:Registered Service com.victronenergy.temperature.mqtt_fe002_t1 under DeviceInstance 2
2024-10-29 12:02:52.624431500 INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 2}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/2', 'R': 'R/028149766bb2/temperature/2', 'W': 'W/028149766bb2/temperature/2'}}} to device/fe002/DBus, status is 0

^C

**Reply by BobvanderWeg** [reply]
I think I might have some results now; do not know what changed but the service is running and the devices are apearing in the GX again:
though I still have a Warning

root@nanopi:/data# more /var/log/dbus-mqtt-devices/current
@400000006720b9dd0256e65c INFO:device_manager:Received device status message {'clientId': 'ev001', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'ev1': 'evcharger'}}
@400000006720b9dd02830b24 INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'ev1': 1}, 'topicPath': {'ev1': {'N': 'N/028149766bb2/evcharger/1', 'R': 'R/028149766bb2/evcharger/1', 'W': 'W/028149766bb2/evcharger/1'}}} to device/ev001/DBus, status is 0
@400000006720b9dd0c49ab04 INFO:device_manager:Received device status message {'clientId': 'fe001', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'t1': 'temperature'}}
@400000006720b9dd0c897edc INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 1}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/1', 'R': 'R/028149766bb2/temperature/1', 'W': 'W/028149766bb2/temperature/1'}}} to device/fe001/DBus, status is 0
@400000006720b9dd0cc251bc INFO:device_manager:Received device status message {'clientId': 'fe002', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'t1': 'temperature'}}
@400000006720b9dd0cf5583c INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 2}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/2', 'R': 'R/028149766bb2/temperature/2', 'W': 'W/028149766bb2/temperature/2'}}} to device/fe002/DBus, status is 0
@400000006720c0e50702f9ac INFO:device_manager:Received device status message {'clientId': 'ev001', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'ev1': 'evcharger'}}
@400000006720c0e5072f590c INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'ev1': 1}, 'topicPath': {'ev1': {'N': 'N/028149766bb2/evcharger/1', 'R': 'R/028149766bb2/evcharger/1', 'W': 'W/028149766bb2/evcharger/1'}}} to device/ev001/DBus, status is 0
@400000006720c0e510d6247c INFO:device_manager:Received device status message {'clientId': 'fe001', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'t1': 'temperature'}}
@400000006720c0e510fdf7cc INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 1}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/1', 'R': 'R/028149766bb2/temperature/1', 'W': 'W/028149766bb2/temperature/1'}}} to device/fe001/DBus, status is 0
@400000006720c0e5111e493c INFO:device_manager:Received device status message {'clientId': 'fe002', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'t1': 'temperature'}}
@400000006720c0e5113f1f7c INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 2}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/2', 'R': 'R/028149766bb2/temperature/2', 'W': 'W/028149766bb2/temperature/2'}}} to device/fe002/DBus, status is 0
@400000006720c7ed0a8da324 INFO:device_manager:Received device status message {'clientId': 'ev001', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'ev1': 'evcharger'}}
@400000006720c7ed0abd07f4 INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'ev1': 1}, 'topicPath': {'ev1': {'N': 'N/028149766bb2/evcharger/1', 'R': 'R/028149766bb2/evcharger/1', 'W': 'W/028149766bb2/evcharger/1'}}} to device/ev001/DBus, status is 0
@400000006720c7ed14829e34 INFO:device_manager:Received device status message {'clientId': 'fe001', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'t1': 'temperature'}}
@400000006720c7ed14ae0394 INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 1}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/1', 'R': 'R/028149766bb2/temperature/1', 'W': 'W/028149766bb2/temperature/1'}}} to device/fe001/DBus, status is 0
@400000006720c7ed14d653e4 INFO:device_manager:Received device status message {'clientId': 'fe002', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'t1': 'temperature'}}
@400000006720c7ed14fcdf14 INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 2}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/2', 'R': 'R/028149766bb2/temperature/2', 'W': 'W/028149766bb2/temperature/2'}}} to device/fe002/DBus, status is 0
@400000006720cb2e00b08d44 *** CCGX booted (0) ***
@400000006720cbe0105c7bf4 *** starting dbus-mqtt-devices ***
@400000006720cbe13a778974 dbus_mqtt_devices v{} 0.8.0
@400000006720cbe13a77ac9c INFO:logger:Loglevel set to INFO
@400000006720cbe13a91ef44 INFO:dbus_mqtt_devices:-------- dbus_mqtt_devices, v0.8.0 is starting up --------
@400000006720cbe2001acd54 INFO:device_manager:Using portalId 028149766bb2
@400000006720cbe22bed00b4 INFO:mqtt_gobject_bridge:[Init] Connecting to local broker
@400000006720cbe22e47a60c INFO:device_manager:[Connected] Result code 0
@400000006720cef334d50064 INFO:device_manager:Received device status message {'clientId': 'ev001', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'ev1': 'evcharger'}}
@400000006720cef334f9ed84 INFO:device:**** Registering device: ev001, services: {'ev1': 'evcharger'} ****
@400000006720cef33503c50c INFO:device:Registering Service evcharger for client ev001
@400000006720cef3350bfe84 INFO:device_service_config:About to open config file
@400000006720cef4271633e4 INFO:device_service:Registering service evcharger for client ev001 at path com.victronenergy.evcharger.mqtt_ev001_ev1
@400000006720cef42d06d90c INFO:settingsdevice:Setting /Settings/Devices/mqtt_ev001_ev1/ClassAndVrmInstance does not exist yet or must be adjusted
@400000006720cef4319adc34 WARNING:vedbus:USING OUTDATED REGISTRATION METHOD!
@400000006720cef431a4dacc WARNING:vedbus:Please set register=False, then call the register method after adding all mandatory paths. See https://github.com/victronenergy/venus/wiki/dbus-api
@400000006720cef431f1260c INFO:vedbus:registered ourselves on D-Bus as com.victronenergy.evcharger.mqtt_ev001_ev1
@400000006720cef4332c3e1c INFO:device_service:Registered Service com.victronenergy.evcharger.mqtt_ev001_ev1 under DeviceInstance 1
@400000006720cef4336e63b4 INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'ev1': 1}, 'topicPath': {'ev1': {'N': 'N/028149766bb2/evcharger/1', 'R': 'R/028149766bb2/evcharger/1', 'W': 'W/028149766bb2/evcharger/1'}}} to device/ev001/DBus, status is 0
@400000006720cef433bad21c INFO:device_manager:Received device status message {'clientId': 'fe001', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'t1': 'temperature'}}
@400000006720cef433c6b514 INFO:device:**** Registering device: fe001, services: {'t1': 'temperature'} ****
@400000006720cef433d0a40c INFO:device:Registering Service temperature for client fe001
@400000006720cef433d9eef4 INFO:device_service_config:About to open config file
@400000006720cef524ab5634 INFO:device_service:Registering service temperature for client fe001 at path com.victronenergy.temperature.mqtt_fe001_t1
@400000006720cef526dee7a4 INFO:settingsdevice:Setting /Settings/Devices/mqtt_fe001_t1/ClassAndVrmInstance does not exist yet or must be adjusted
@400000006720cef52ae521ec WARNING:vedbus:USING OUTDATED REGISTRATION METHOD!
@400000006720cef52af035dc WARNING:vedbus:Please set register=False, then call the register method after adding all mandatory paths. See https://github.com/victronenergy/venus/wiki/dbus-api
@400000006720cef52b7ba144 INFO:vedbus:registered ourselves on D-Bus as com.victronenergy.temperature.mqtt_fe001_t1
@400000006720cef52cfb4434 INFO:device_service:Registered Service com.victronenergy.temperature.mqtt_fe001_t1 under DeviceInstance 1
@400000006720cef52d2ef2ac INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 1}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/1', 'R': 'R/028149766bb2/temperature/1', 'W': 'W/028149766bb2/temperature/1'}}} to device/fe001/DBus, status is 0
@400000006720cef52d7b361c INFO:device_manager:Received device status message {'clientId': 'fe002', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'t1': 'temperature'}}
@400000006720cef52d87828c INFO:device:**** Registering device: fe002, services: {'t1': 'temperature'} ****
@400000006720cef52d8f0c3c INFO:device:Registering Service temperature for client fe002
@400000006720cef52d96b914 INFO:device_service_config:About to open config file
@400000006720cef61d58e6bc INFO:device_service:Registering service temperature for client fe002 at path com.victronenergy.temperature.mqtt_fe002_t1
@400000006720cef61f4c2754 INFO:settingsdevice:Setting /Settings/Devices/mqtt_fe002_t1/ClassAndVrmInstance does not exist yet or must be adjusted
@400000006720cef623dc4664 WARNING:vedbus:USING OUTDATED REGISTRATION METHOD!
@400000006720cef623e44d14 WARNING:vedbus:Please set register=False, then call the register method after adding all mandatory paths. See https://github.com/victronenergy/venus/wiki/dbus-api
@400000006720cef6243d3e9c INFO:vedbus:registered ourselves on D-Bus as com.victronenergy.temperature.mqtt_fe002_t1
@400000006720cef624e09704 INFO:device_service:Registered Service com.victronenergy.temperature.mqtt_fe002_t1 under DeviceInstance 2
@400000006720cef62538118c INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 2}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/2', 'R': 'R/028149766bb2/temperature/2', 'W': 'W/028149766bb2/temperature/2'}}} to device/fe002/DBus, status is 0
@400000006720d2273b9303c4 INFO:device_manager:Received device status message {'clientId': 'fe001', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'t1': 'temperature'}}
@400000006720d2280022e78c INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 1}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/1', 'R': 'R/028149766bb2/temperature/1', 'W': 'W/028149766bb2/temperature/1'}}} to device/fe001/DBus, status is 0
@400000006720d2280043c59c INFO:device_manager:Received device status message {'clientId': 'fe002', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'t1': 'temperature'}}
@400000006720d228006b141c INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 2}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/2', 'R': 'R/028149766bb2/temperature/2', 'W': 'W/028149766bb2/temperature/2'}}} to device/fe002/DBus, status is 0
@400000006720d22819c96ecc INFO:device_service:Value changed, path: /Position, value: Alfen BV, setting: Position
@400000006720d3170ec2f034 INFO:device_manager:Received device status message {'clientId': 'fe001', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'t1': 'temperature'}}
@400000006720d3170efb981c INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 1}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/1', 'R': 'R/028149766bb2/temperature/1', 'W': 'W/028149766bb2/temperature/1'}}} to device/fe001/DBus, status is 0
@400000006720d31812bbdc3c INFO:device_manager:Received device status message {'clientId': 'fe002', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'t1': 'temperature'}}
@400000006720d318133f6034 INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'t1': 2}, 'topicPath': {'t1': {'N': 'N/028149766bb2/temperature/2', 'R': 'R/028149766bb2/temperature/2', 'W': 'W/028149766bb2/temperature/2'}}} to device/fe002/DBus, status is 0
@400000006720d31e106eac34 INFO:device_manager:Received device status message {'clientId': 'gr001', 'connected': 0, 'version': 'v1.0 ALPHA', 'services': {'gr1': 'pvinverter'}}
@400000006720d31e107ac1f4 WARNING:device_manager:tried to remove device gr001 that is not registered
@400000006720d32a3abe98dc INFO:device_manager:Received device status message {'clientId': 'ev001', 'connected': 1, 'version': 'v1.0 ALPHA', 'services': {'ev1': 'evcharger'}}
@400000006720d32a3aea7754 INFO:device_manager:publish {'portalId': '028149766bb2', 'deviceInstance': {'ev1': 1}, 'topicPath': {'ev1': {'N': 'N/028149766bb2/evcharger/1', 'R': 'R/028149766bb2/evcharger/1', 'W': 'W/028149766bb2/evcharger/1'}}} to device/ev001/DBus, status is 0

**Comment by BobvanderWeg** [comment]
@400000006720dd0c1cfbf1dc     from device_manager import MQTTDeviceManager
@400000006720dd0c1cfbf9ac   File "/data/drivers/dbus-mqtt-devices-0.8.0/device_manager.py", line 17, in <module>
@400000006720dd0c1cfc094c     import yaml
@400000006720dd0c1cfc111c ModuleNotFoundError: No module named 'yaml'
@400000006720dd0c267826f4 *** starting dbus-mqtt-devices ***
@400000006720dd0d378cb5f4 Traceback (most recent call last):
@400000006720dd0d378cdd04   File "/data/drivers/dbus-mqtt-devices-0.8.0/dbus_mqtt_devices.py", line 34, in <module>
@400000006720dd0d378ceca4     from device_manager import MQTTDeviceManager
@400000006720dd0d378cf85c   File "/data/drivers/dbus-mqtt-devices-0.8.0/device_manager.py", line 17, in <module>
@400000006720dd0d378d0414     import yaml
@400000006720dd0d378d07fc ModuleNotFoundError: No module named 'yaml'
@400000006720dd0e05df0f34 *** starting dbus-mqtt-devices ***

**Comment by freakent** [comment]
I'm very concerned by this message. 

@400000006720cef4319adc34 WARNING:vedbus:USING OUTDATED REGISTRATION METHOD!
@400000006720cef431a4dacc WARNING:vedbus:Please set register=False, then call the register method after adding all mandatory paths. See https://github.com/victronenergy/venus/wiki/dbus-api

It looks like Victron are making some potentially breaking changes to the api.  I'll have to investigate this.
